core/src/main/java/org/mule/retry/policies/AbstractPolicyTemplate.java

```diff
@@ -100,13 +100,14 @@ public RetryContext execute(RetryCallback callback, WorkManager workManager) thr
             }
             while (status.isOk());
 
+            context.setFailed(cause);
```

Comment:
When InterruptedException or InterruptedIOException are triggered, they are catched by the AbstractPolicy that after returns a RetryContext. In these scenarios, the code caller doesn't have any way to propagate the exception or to have information about this.